### PR TITLE
IGNITE-19979 Table start might fail due to an outdated causality token

### PR DESCRIPTION
**Pull request by rpuch**
https://issues.apache.org/jira/browse/IGNITE-19979

### CODE REVIEW

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java [hunk 1]
```diff
@@ -909,27 +909,27 @@ private CompletableFuture<?> createTablePartitionsLocally(
...
-                    return failedFuture(e);
-                }
+        return assignmentsUpdatedVv.update(causalityToken, (token, e) -> {
+            if (e != null) {
+                return failedFuture(e);
+            }
 
-                return updateAssignmentsFuture;
-            });
+            return localPartsUpdateFuture.thenCompose(unused ->
```
Comment (ibessonov) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java:929]: Assuming that `assignmentsUpdatedVv` depends on `localPartsByTableIdVv`, I think it's safe to assume that this particular `thenCompose` is redundant.

Comment (ibessonov) [reply modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java:929]: We could also assert that `tablesByIdVv.get(causalityToken)` is already completed at this point

Comment (rpuch) [reply modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java:929]: If I change the code to the following

```
        return assignmentsUpdatedVv.update(causalityToken, (token, e) -> {
            if (e != null) {
                return failedFuture(e);
            }

            return tablesByIdVv.get(causalityToken)
                    .thenComposeAsync(tablesById -> inBusyLock(busyLock, updateAssignmentsClosure), ioExecutor);
        });
```
, I get an assertion:

```
Caused by: java.lang.AssertionError
	at org.apache.ignite.internal.table.distributed.TableManager.getPartitionStorages(TableManager.java:2502)
	at org.apache.ignite.internal.table.distributed.TableManager.lambda$createTablePartitionsLocally$13(TableManager.java:763)
	at org.apache.ignite.internal.util.IgniteUtils.inBusyLock(IgniteUtils.java:893)
	at org.apache.ignite.internal.table.distributed.TableManager.lambda$createTablePartitionsLocally$17(TableManager.java:930)
	at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1072)
	... 4 more
```

Also, `tablesByIdVv.get(causalityToken)` is not done when `assignmentsUpdatedVv`-related callback is called, so the assertion cannot be added.

Comment (rpuch) [reply modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java:929]: Looks like VV.complete() is not guaranteed to be called strictly after all VV.update-related futures complete

**Review by ibessonov — APPROVED** [review]

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java (hunk 1 above)
Comment (ibessonov) [reply modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java:929]: Ok, considering that you're fixing a flaky test, this is urgent, so let's investigate the assertion later

**Pull request merged**